xmodule/seq_block.py

```diff
@@ -376,8 +376,18 @@ def get_metadata(self, view=STUDENT_VIEW, context=None):
         meta['display_name'] = self.display_name_with_default
         meta['format'] = getattr(self, 'format', '')
         meta['is_hidden_after_due'] = is_hidden_after_due
+        meta['navigation_disabled'] = self.is_sequence_navigation_disabled()
         return meta
 
+    def is_sequence_navigation_disabled(self):
+        """
+        Returns whether the navigation to other sequences is disabled.
+
+        This is used to disable the navigation to other sequences when the
+        current sequence is configured as Hide from Table of Contents.
+        """
+        return getattr(self, "hide_from_toc", False)
```

Comment:
Note to self: add docstring here explaining that other types of validations could be added in the future.